astroid/brain/brain_dataclasses.py

```diff
@@ -231,19 +232,20 @@ def infer_dataclass_attribute(
 
 
 def infer_dataclass_field_call(
-    node: AssignName, ctx: context.InferenceContext = None
+    node: Call, ctx: context.InferenceContext = None
 ) -> Generator:
     """Inference tip for dataclass field calls."""
-    field_call = node.parent.value
-    default_type, default = _get_field_default(field_call)
-    if not default_type:
-        yield Uninferable
-    elif default_type == "default":
-        yield from default.infer(context=ctx)
-    else:
-        new_call = parse(default.as_string()).body[0].value
-        new_call.parent = field_call.parent
-        yield from new_call.infer(context=ctx)
+    if isinstance(node.parent, (AnnAssign, Assign)):
```

Comment:
```suggestion
    if not isinstance(node.parent, (AnnAssign, Assign)):
        return
```
Unless it does not works with yield ? It permits to simplify the number of things to keep in mind when reading the code and reduce the diff and future conflict.